[FEATURE] [MER-4617] [MER-4618] Audit log feature

darrensiegel opened the pull request:
Implements the Audit Log feature (core infrastructure, Admin UI, initial usages for account deletion and course project publication)

nicocirio reviewed (changes requested):
Nice work! I tested it locally and it works as expected. 
On the liveview we do not have the option to filter by "system" author. I believe that would be a nice-to-have filter.
On the other hand, I noticed we added many event_types that we are not yet auditing.

darrensiegel commented:
@nicocirio I've addressed your review comments:
- Fixed it to rely on the router scoping
- Removed all of the "not yet implemented" event types
- Remove the "not yet used" dynamic filters

I didn't implement the "filter by system" idea yet - as we do not have any events yet that are strictly "system" driven. But that is a good idea and I captured it as a comment next to the event_types definition.

nicocirio commented:
> @nicocirio I've addressed your review comments:
> 
> * Fixed it to rely on the router scoping
> * Removed all of the "not yet implemented" event types
> * Remove the "not yet used" dynamic filters
> 
> I didn't implement the "filter by system" idea yet - as we do not have any events yet that are strictly "system" driven. But that is a good idea and I captured it as a comment next to the event_types definition.

Great, thanks! One last thing, we can now remove the author system role check on the LiveView mount callback, since now that is checked in the router.
